src/main/java/ch/hsr/epj/localshare/demo/network/discovery/searcher/NetworkDiscovery.java

```diff
@@ -9,7 +9,7 @@ public void startSearchProcess() {
         IPAddressUtil.generateIPsInNetmask(IPAddressUtil.getLocalIPAddress());
 
     Statemachine discovery = new Statemachine();
-    discovery.addListOfIPsToScan(listOfIPsToProbe);
+    Statemachine.addListOfIPsToScan(listOfIPsToProbe);
```

Comment:
Does this work as intended? Some parts of the Statemachine class are static, but the state member for example is not.

Reply:
Method `addListOfIPsToScan()` is a static method and attached to the class (exists only once for all instances of class). Should only be called directly on class `Statemachine`. `discovery` is a instance of class `Statemachine`.